### PR TITLE
Update polling logic for long-running operations

### PR DESCRIPTION
**Pull request by garimakhulbe**
Handle polling for both Asynchronous and Location headers
Add cancel channel argument to cancel polling

### CODE REVIEW

**Comment by ahmetalpbalkan** [comment]
@garimakhulbe thanks! I'll try this tomorrow.

**Comment by garimakhulbe** [comment]
Okay. Finger-crossed. :)

**Comment by ahmetalpbalkan** [comment]
@garimakhulbe works great, LGTM. Can you please also update changelog with latest changes merged to `storage`? Thanks.

**Comment by brendandixon** [comment]
@garimakhulbe @ahmetalpbalkan Let's fix the polling delay before merging. It's a quick fix. :)

**Comment by garimakhulbe** [comment]
UPdated. Check now.

File: arm/CHANGELOG.md [hunk 1]
```diff
@@ -2,6 +2,11 @@
 
 -----
 
+## `v2.1.0-beta`
+
+- Uses go-autorest v7.1.0 (Polling logic changed).
+- Cancel channel argument added in long-running calls to cancel polling if needed.
```
Comment (ahmetalpbalkan) [inline arm/CHANGELOG.md]: Can you please add latest storage changes merged  as well?

**Comment by brendandixon** [comment]
@garimakhulbe To add the latest changes, first `git fetch upstream`, then `git rebase -i upstream/master`. Finally, `git push -f` your code to update the PR. That should do it!

**Comment by garimakhulbe** [comment]
All updated. Take a look. Will tag SDK after merge.

File: CHANGELOG.md [hunk 2]
```diff
@@ -2,6 +2,14 @@
 
 -----
 
+## `v2.1.0-beta`
+
+- Uses go-autorest v7.0.2 (Polling logic changed).
+- Cancel channel argument added in long-running calls to cancel polling if needed.
+- Allow caller to provide headers for DeleteBlob methods.
```
Comment (ahmetalpbalkan) [inline CHANGELOG.md]: can you please group these under `storage`? it's vague that where they belong. In the future you can also provide better short descriptions. Take a look at this:

```
- storage: Allow caller to provide headers for DeleteBlob methods.
- storage: Enables connection sharing with http keepalive
```

**Comment by ahmetalpbalkan** [comment]
@garimakhulbe please don't merge. This version doesn't even poll at all. Sent an email.

**Comment by brendandixon** [comment]
@ahmetalpbalkan We're going to need data. First, coding precisely to the specification has not worked. Then coding to the packets you sent is now failing. We need to see precisely the data returned so we know how to react. As it stands, the code is well-tested and does poll for 201 / 202 is the Azure-AsyncHeader is present. Something is amiss...

**Comment by ahmetalpbalkan** [comment]
LGTM let's merge.

**Pull request merged**